sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -134,8 +134,87 @@ template <typename Arg0, typename... Args>
 void printArgs(Arg0 arg0, Args... args) {
   std::cout << "       ";
   print(arg0);
-  printArgs(std::forward<Args>(args)...);
+  pi::printArgs(std::forward<Args>(args)...);
 }
+
+// Utility function to check return from pi calls.
+// Throws if pi_result is not a PI_SUCCESS.
+template <typename Exception = cl::sycl::runtime_error>
+inline void checkPiResult(PiResult pi_result) {
+  CHECK_OCL_CODE_THROW(pi_result, Exception);
+}
+
+// Class to call PI API, trace and get the result.
+// To Trace : Set SYCL_PI_TRACE environment variable.
+// Template Arguments:
+//    FnType  - Type of Function pointer to the PI API.
+//    FnOffset- Offset to the Function Pointer in the piPlugin::FunctionPointers
+//    structure. Used to differentiate between APIs with same pointer type,
+//    E.g.: piDeviceRelease and piDeviceRetain. Differentiation needed to avoid
+//    redefinition error during explicit specialization of class in pi.cpp.
+// Members: Initialized in default constructor in Class Template Specialization.
+// Usage:
+// Operator() - Call, Trace and Get result
+// Use Macro PI_CALL_NOCHECK call the constructor directly.
+template <typename FnType, size_t FnOffset> class CallPi {
```

Comment:
I don't fully understand why these changes are present in the diff, is it a result of not fully correct merge from `sycl` branch?